### PR TITLE
Contrib/create package versioning

### PR DESCRIPTION
**Pull request by tweak-wtf**
## Description
This Draft changes the way services are built. It follows the approach used in `ayon-shotgrid`.

This branch has the changes from `develop/production-ready` already merged in. That's why there are so many changed files.

The new `Makefile` will query the current package version and update the respective service `pyproject.toml` using `sed`.
Currently this is only supported on Linux and would need a port to `manage.ps1`.
`docker-compose.yml` files have been removed in favor for `make dev` and `make bash` commands to quickly test the built images. ASH will create single containers from an image but no stack from a compose file.

It replaces the `version.py` approach in favor of new `package.py` approach. This would need to be reviewed by AYON tho. Tried my best to remove any code references to deprecated approach but not sure if there is more code that can be safely removed.

## Testing Steps
- navigate to `./services`
- `make build SERVICE=leecher`
- `make build SERVICE=processor`
- `make dev SERVICE=leecher`
- `make dev SERVICE=processor`

### CODE REVIEW

**Comment by ymoriaud** [comment]
Thanks !
FYI, I will review it later this week.

**Review by ymoriaud — COMMENTED** [review]
Just added one suggestion and a question. :)
Thanks @tweak-wtf for this PR !

File: server/__init__.py [hunk 1]
```diff
@@ -12,32 +12,33 @@
...
 
 
 from .vendors.aquarium import Aquarium, DEFAULT_STATUSES
 
-try:
-    from .version import __version__ # type: ignore
-except ModuleNotFoundError:
-    # temporary solution for local development
-    # version is pushed from package.py in ayon >= 1.0.3
-    __version__ = "0.0.0"
```
[the remark this answers is not in the document]
Comment (tweak-wtf) [reply server/__init__.py:40]: it shouldn't cause any harm.
i removed it to comply with ynput's new `package.py` workflow.

File: services/Makefile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+AYON_ADDON_NAME=$(shell python -c "import os;import sys;content={};f=open('$(CURDIR)/../package.py');exec(f.read(),content);f.close();print(content['name'])")
+AYON_ADDON_VERSION=$(shell python -c "import os;import sys;content={};f=open('$(CURDIR)/../package.py');exec(f.read(),content);f.close();print(content['version'])")
+SERVICE = $(error Please specify the service to build with 'SERVICE', for example: 'make build SERVICE=leecher')
+IMAGE := ynput/ayon-aquarium-$(SERVICE):$(AYON_ADDON_VERSION)
+
+
+build:
+	cp -r ./common/ ./$(SERVICE)/aquarium_common
+	sed -i '/^\[tool.poetry\]/,/^\[/ s/^version = .*/version = "$(AYON_ADDON_VERSION)"/' $(SERVICE)/pyproject.toml
+	docker build -t $(IMAGE) ./$(SERVICE)
```
[the remark this answers is not in the document]
Comment (tweak-wtf) [reply services/Makefile]: yeah this continues to work under Linux.
addressed in this commit https://github.com/ynput/ayon-aquarium/commit/473fdb798dab3ea8b8a861cb05564cd1cadba83e

**Review by ymoriaud — APPROVED** [review]
LGTM, thanks !

**Pull request merged**